javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistMethodDeclaration.java

```diff
@@ -160,7 +157,7 @@ public boolean isAbstract() {
     public List<ResolvedTypeParameterDeclaration> getTypeParameters() {
         try {
             if (ctMethod.getGenericSignature() == null) {
-                return Collections.emptyList();
+                return new ArrayList<>();
```

Comment:
Why is that?

Reply:
Yes, the list is modified in the class MethodResolutionLogic, line 271.